src/compiler/binder.ts

```diff
@@ -661,7 +661,11 @@ module ts {
         }
 
         function bindExportAssignment(node: ExportAssignment) {
-            if (node.expression.kind === SyntaxKind.Identifier) {
+            if (!container.symbol || !container.symbol.exports) {
+                // Export assignment in some sort of block construct
+                bindAnonymousDeclaration(node, SymbolFlags.Alias, "");

```

Comment:
can you use a dunder name instead?  I'm not very comfortable with entities having empty names.